File: frontend/src/api/checkoutApi.js
import axiosClient from './axiosClient';

const checkoutApi = {
  /**
   * Tạo booking + khởi tạo thanh toán.
   *
   * @param {{
   *   propertySlug: string;
   *   branchCode: string;
   *   detailSlug: string;
   *   checkIn: string;
   *   checkOut: string;
   *   guests?: string;
   *   guestName: string;
   *   guestPhone: string;
   *   guestEmail: string;
   *   specialNote?: string;
   *   promoCode?: string | null;
   *   paymentMethod: 'card' | 'momo' | 'bank' | 'wallet' | 'cherry_wallet';
   * }} payload
   */
  async startPay(payload) {
    const res = await axiosClient.post('/checkout/pay', payload);
    return res.data?.data ?? res.data;
  },

  async getStatus(bookingCode) {
    const res = await axiosClient.get(`/checkout/status/${encodeURIComponent(bookingCode)}`);
    return res.data?.data ?? res.data;
  },

  /** Xác thực redirect VNPay — truyền toàn bộ query string */
  async verifyVnpay(queryParams) {
    const res = await axiosClient.get('/checkout/verify/vnpay', { params: queryParams });
    return res.data?.data ?? res.data;
  },

  /** Xác thực redirect MoMo */
  async verifyMomo(queryParams) {
    const res = await axiosClient.get('/checkout/verify/momo', { params: queryParams });
    return res.data?.data ?? res.data;
  },
};

export default checkoutApi;

File: mobile/lib/api/checkout_api.dart
import 'api_client.dart';

class CheckoutApi {
  CheckoutApi(this._client);

  final ApiClient _client;

  Future<Map<String, dynamic>> startPay(Map<String, dynamic> payload) async {
    final json = await _client.post('/checkout/pay', payload);
    return json['data'] as Map<String, dynamic>;
  }

  Future<Map<String, dynamic>> getStatus(String bookingCode) async {
    final json = await _client.get('/checkout/status/${Uri.encodeComponent(bookingCode)}');
    return json['data'] as Map<String, dynamic>;
  }

  Future<Map<String, dynamic>> verifyVnpay(Map<String, String> query) async {
    final json = await _client.get('/checkout/verify/vnpay', query: query);
    return json['data'] as Map<String, dynamic>;
  }

  Future<Map<String, dynamic>> verifyMomo(Map<String, String> query) async {
    final json = await _client.get('/checkout/verify/momo', query: query);
    return json['data'] as Map<String, dynamic>;
  }
}
